File: algorithms/2_medium/D/02849_determine_if_a_cell_is_reachable_at_a_given_time/solution.py
class Solution:
    def isReachableAtTime(self, sx: int, sy: int, fx: int, fy: int, t: int) -> bool:
        dx = abs(fx - sx)
        dy = abs(fy - sy)

        if dx == 0 and dy == 0 and t == 1:
            return False

        return max(dx, dy) <= t

File: algorithms/2_medium/D/02849_determine_if_a_cell_is_reachable_at_a_given_time/solution.rs
impl Solution {
    pub fn is_reachable_at_time(sx: i32, sy: i32, fx: i32, fy: i32, t: i32) -> bool {
        let dx = (fx - sx).abs();
        let dy = (fy - sy).abs();

        if dx == 0 && dy == 0 && t == 1 {
            return false;
        }

        dx.max(dy) <= t
    }
}
